gridstatus/lmp_config.py

```diff
@@ -0,0 +1,211 @@
+        if len(args) == 0:
+            if (
+                "date" not in kwargs and "start" not in kwargs
+            ) or "market" not in kwargs:
+                raise ValueError("date/start and market are required")
+            if "date" in kwargs:
+                date_field = "date"
+            else:
+                date_field = "start"
+            date = kwargs[date_field]
+            market = kwargs["market"]
+        elif len(args) == 1:
+            if ("date" in kwargs or "start" in kwargs) and "market" in kwargs:
+                if "date" in kwargs:
+                    date_field = "date"
+                else:
+                    date_field = "start"
+                date = kwargs[date_field]
+                market = kwargs["market"]
+            elif "date" in kwargs:
+                date_field = "start"
+                date = kwargs[date_field]
+                market = args[0]
+            elif "market" in kwargs:
+                date = args[0]
+                market = kwargs["market"]
+            else:
+                raise ValueError("Missing date or market")
+        else:
+            date = args[0]
+            market = args[1]
+
+        original_date_arg = date
+        date = self._parse_date(date, tz=tz)
+        market = self._parse_market(market)
+
+        self._check_support(original_date_arg, date, market, tz)
+
+        modify_date_arg = original_date_arg != "latest"
+
+        if len(args) == 0:
+            if modify_date_arg:
+                kwargs[date_field] = date
+            kwargs["market"] = market
+        elif len(args) == 1:
+            if date_field in kwargs and "market" not in kwargs:
+                if modify_date_arg:
+                    kwargs[date_field] = date
+                args[0] = market
+            elif date_field not in kwargs and "market" in kwargs:
+                if modify_date_arg:
+                    args[0] = date
+                kwargs["market"] = market
+        else:
+            if modify_date_arg:
+                args[0] = date
+            args[1] = market
+
+        return {
+            "args": tuple(args),
+            "kwargs": kwargs,
+        }
+
+    def _check_support(self, original_date_arg, date_arg, market_arg, tz):
+        if market_arg not in self.supports:
+            raise NotSupported(f"{market_arg} not supported")
+
+        if original_date_arg in (
+            "latest",
+            "today",
+        ):
+            supported = original_date_arg in self.supports[market_arg]
+        elif lmp_config.__is_today(date_arg, tz):
+            supported = "today" in self.supports[market_arg]
+        else:
+            supported = "historical" in self.supports[market_arg]
+
+        if not supported:
+            raise NotSupported(
+                f"{market_arg} does not support {repr(original_date_arg)}",
+            )
+
+    # copied from utils.py to avoid circular import
+    @staticmethod
+    def __handle_date(date, tz=None):
```

Comment:
you can avoid a circular import putting the import inside the method call where you use it. that way you do have to copy code

Reply:
Ah, thanks